### PR TITLE
Alerting: Add message options for Telegram contact point

### LINKED ISSUE #68167: Alerting: Telegram contact point options
**What is the chore?**:
Several [parameters](https://core.telegram.org/bots/api#sendmessage) for the Telegram API's `sendMessage` method are unable to be configured in Grafana when using Telegram as a contact point for alerts. Useful ones to be able to configure would be `disable_web_page_preview` and `protect_content`.

### PR DESCRIPTION
**Pull request by nutmos**
**What is this feature?**

Chore: Add `disable_web_page_preview` and `protect_content` options for the Telegram contact point option.

**Why do we need this feature?**

Add the useful options for the Telegram message when using as a contact point.

**Who is this feature for?**

Grafana users that use Telegram as a contact point.

**Which issue(s) does this PR fix?**:

Fixes #68167
Depends on https://github.com/grafana/alerting/pull/124

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts [hunk 1]
```diff
@@ -377,6 +377,12 @@ export const cloudNotifierTypes: Array<NotifierDTO<CloudNotifierType>> = [
       option('message', 'Message', 'Message template', {
         placeholder: '{{ template "webex.default.message" .}}',
       }),
+      option('disable_web_page_preview', 'Disable web page preview', 'Disable web page preview in telegram messages', {
+        element: 'checkbox',
+      }),
+      option('protect_content', 'Protect content', 'Disable forward and save messages on telegram', {
+        element: 'checkbox',
+      }),
```
Comment (yuri-tceretian) [inline public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts:385]: My understanding that these are options of the cloud Alertmanager, that uses the notifiers of Prometheus Alertmanager. Afaiu, you did not change it, did you?
UI for Grafana-managed alerts is automatically built upon changes you made in the .go file.

**Comment by yuri-tceretian** [comment]
It is blocked by https://github.com/grafana/alerting/pull/124 which needs to be merged first and then we need to update the module in this PR.

File: pkg/services/ngalert/notifier/channels_config/available_channels.go [hunk 2]
```diff
@@ -843,6 +843,18 @@ func GetAvailableNotifiers() []*NotifierPlugin {
 					Description:  `Mode for parsing entities in the message text. Default is 'HTML'`,
 					PropertyName: "parse_mode",
 				},
+				{
+					Label:        "Disable Web Page Preview",
+					Description:  "Sends the message without showing the web page preview.",
```
Comment (santihernandezc) [inline pkg/services/ngalert/notifier/channels_config/available_channels.go:848]: We can use the descriptions from Telegram's [docs](https://core.telegram.org/bots/api):
```suggestion
					Description:  "Disables link previews for links in this message",
```

File: pkg/services/ngalert/notifier/channels_config/available_channels.go [hunk 3]
```diff
@@ -843,6 +843,18 @@ func GetAvailableNotifiers() []*NotifierPlugin {
...
+					Element:      ElementTypeCheckbox,
+					PropertyName: "disable_web_page_preview",
+				},
+				{
+					Label:        "Protect Content",
+					Description:  "Messages will be protected from forwarding and saving.",
```
Comment (santihernandezc) [inline pkg/services/ngalert/notifier/channels_config/available_channels.go:854]: Same here
```suggestion
					Description:  "Protects the contents of the sent message from forwarding and saving",
```

File: public/app/features/alerting/unified/mockGrafanaNotifiers.ts [hunk 4]
```diff
@@ -1539,6 +1539,40 @@ export const grafanaAlertNotifiersMock: NotifierDTO[] = [
         secure: false,
         dependsOn: '',
       },
+      {
+        element: 'checkbox',
+        inputType: '',
+        label: 'Disable Web Page Preview',
+        description: 'Sends the message without showing the web page preview.',
```
Comment (santihernandezc) [inline public/app/features/alerting/unified/mockGrafanaNotifiers.ts:1546]:
```suggestion
        description: 'Disables link previews for links in this message',
```

File: public/app/features/alerting/unified/mockGrafanaNotifiers.ts [hunk 5]
```diff
@@ -1539,6 +1539,40 @@ export const grafanaAlertNotifiersMock: NotifierDTO[] = [
...
+        required: false,
+        validationRule: '',
+        secure: false,
+        dependsOn: '',
+      },
+      {
+        element: 'checkbox',
+        inputType: '',
+        label: 'Protect Content',
+        description: 'Messages will be protected from forwarding and saving.',
```
Comment (santihernandezc) [inline public/app/features/alerting/unified/mockGrafanaNotifiers.ts:1563]:
```suggestion
        description: 'Protects the contents of the sent message from forwarding and saving',
```

File: public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts (hunk 1 above)
Comment (santihernandezc) [reply public/app/features/alerting/unified/utils/cloud-alertmanager-notifier-types.ts:385]: Ditto @yuri-tceretian, this is for Prometheus' Alertmanager's notifiers only

**Review by santihernandezc — CHANGES_REQUESTED** [review]
Great! Just a few small changes and it'll be good to go.

**Comment by nutmos** [comment]
Thank you for your suggestions. I resolved everything in my branch already. @santihernandezc @yuri-tceretian

**Comment by santihernandezc** [comment]
@nutmos I just pulled your branch and the UI changes look good, but we're still missing the change @yuri-tceretian [mentioned](https://github.com/grafana/grafana/pull/74635#issuecomment-1714258805):
> ...we need to update the module in this PR

I reviewed and merged your [grafana/alerting PR](https://github.com/grafana/alerting/pull/124) yesterday, so now we have to update `go.mod` to include the latest version of `grafana/alerting` [here](https://github.com/nutmos/grafana/blob/e3f182ff1470fbe0eedc42074c412d95be38f41f/go.mod#L62).

**Comment by nutmos** [comment]
@santihernandezc I updated the file from your recommendation already. Thank you.

**Review by santihernandezc — APPROVED** [review]
LGTM 👍 
I tested this using Telegram's mobile app. The protected messages can't be forwarded, and the link previews can be disabled.
Let's wait for @yuri-tceretian's review before merging.

**Review by yuri-tceretian — APPROVED** [review]
LGTM! Tested and it works as expected.

**Pull request merged**